active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+
+### XAML
+
+a user input control with the input bound:
+
+```xaml
+<TextBox Name="ValidationTextbox" 
+         Text="{x:Bind  ViewModel.Person.Name} "/>
+
+```
+#### Code - ViewModel
+
+The “ViewModel” property is the View Model for the Page on which this TextBox is displayed (this.ViewModel).
+``` csharp
+
+public class ViewModel
+{
+    public ViewModel()
+    {
+        Person = new Person();
+    }
+
+    public Person Person { get; set; }
+
+}
+```
+
+#### C# only - Person (Model)
+Person.Name is the source of the bind (this.ViewModel.Person.Name).  Person derives from ValidationWrapper (app code) which will provide the data validation for Name.  It’s marked with [MinLength] and [MaxLength] data annotations.  The property accessors use GetValue/SetValue in the base ValidationWrapper class for storage and to check the validity.
+
+``` csharp
+using System.ComponentModel.DataAnnotations;
+    
+public class Person : ValidationWrapper
+{
+    [MinLength(4, ErrorMessage = "Name must be at least 4 characters")]
+    [MaxLength(6, ErrorMessage = "Name must be no more than 6 characters")]
+    public string Name
+    {	
+        get { return GetValue<string>("Name"); }
+        set { SetValue("Name", value); }
+    }
+}
+```
+
+#### Code – INotifyDataErrorInfo base class
+The NotifyDataErrorInfoBase class (app code) provides a reusable implementation of INotifyDataErrorInfo (new API).  It’s the base of ValidationWrapper, which is the base class for Person.
+``` csharp
+public class NotifyDataErrorInfoBase : INotifyDataErrorInfo
+{
+    protected readonly Dictionary<string, ObservableCollection<ValidationError>> Errors;
+
+    protected NotifyDataErrorInfoBase()
+    {
+        Errors = new Dictionary<string, List<string>>();
+    }
+
+    public bool HasErrors => Errors.Any();
+
+    public event EventHandler<DataErrorsChangedEventArgs> ErrorsChanged;
+
+    public ObservableCollection<ValidationError> GetErrors(string propertyName)
+    {
+        return propertyName != null && Errors.ContainsKey(propertyName)
+            ? Errors[propertyName]
+            : new ObservableCollection<ValidationError>();
+    }
+
+    protected virtual void OnErrorsChanged(string propertyName)
+    {
+        ErrorsChanged?.Invoke(this, new DataErrorsChangedEventArgs(propertyName));
+    }
+
+    protected void ClearErrors()
+    {
+        foreach (var propertyName in Errors.Keys.ToList())
+        {
+            Errors.Remove(propertyName);
+            OnErrorsChanged(propertyName);
+        }
+    }
+}
+```
+#### Code – ValidationWrapper (no new APIs)
+ValidationWrapper derives from the above NotifyDataErrorInfoBase, and provides GetValue/SetValue methods for storage of subclass’ properties.  The SetValue performs validation.  This also provides an implementation of INotifyPropertyChanged.
```

Comment:
Saying "This also provides an implementation of INotifyPropertyChanged." implies this implements this directly, rather than inheriting it from the base class. I suggest removing this sentence to avoid confusion.